### PR TITLE
Draft: revert Draft WB changes from #13294

### PR DESCRIPTION
**Pull request by Roy-043**
Revert the Draft WB related changes of #13294.

### CODE REVIEW

File: src/Mod/Draft/draftfunctions/upgrade.py [hunk 1]
```diff
@@ -227,7 +227,7 @@ def makeShell(objectslist):
                 # of downgrade, nor do they have the same hashCode().
                 # Nevertheless, they still keep reference to their original
                 # colors, capture that in facecolors.
```
Comment (davesrocketshop) [inline src/Mod/Draft/draftfunctions/upgrade.py:229]: ShapeAppearance has replaced ShapeColor. This reversion, although just in a comment, is incorrect.

File: src/Mod/Draft/draftutils/gui_utils.py [hunk 2]
```diff
@@ -474,7 +474,7 @@ def format_object(target, origin=None):
 
     If construction mode is active target is then placed in the construction
     group and the `constr` color is applied to its applicable color properties:
```
Comment (davesrocketshop) [inline src/Mod/Draft/draftutils/gui_utils.py:476]: Again, incorrect

File: src/Mod/Draft/draftutils/utils.py [hunk 3]
```diff
@@ -88,14 +88,14 @@ def get_default_shape_style():
...
     return {
-        "DisplayMode":     ("index", display_mode_index, DISPLAY_MODES[display_mode_index]),
-        "DrawStyle":       ("index", draw_style_index, DRAW_STYLES[draw_style_index]),
-        "LineColor":       ("color", params.get_param_view("DefaultShapeLineColor")),
-        "LineWidth":       ("int",   params.get_param_view("DefaultShapeLineWidth")),
-        "PointColor":      ("color", params.get_param_view("DefaultShapeVertexColor")),
-        "PointSize":       ("int",   params.get_param_view("DefaultShapePointSize")),
-        "ShapeColor":      ("color", params.get_param_view("DefaultShapeColor")),
-        "Transparency":    ("int",   params.get_param_view("DefaultShapeTransparency"))
+        "DisplayMode":  ("index", display_mode_index, DISPLAY_MODES[display_mode_index]),
```
Comment (davesrocketshop) [inline src/Mod/Draft/draftutils/utils.py:91]: This is a formatting change only. Although not incorrect it is unnecessary

File: src/Mod/Draft/draftviewproviders/view_layer.py [hunk 4]
```diff
@@ -70,16 +70,16 @@ def set_override_options(self, vobj, properties):
                              _tip)
             vobj.OverrideLineColorChildren = True
 
-        if "OverrideShapeAppearanceChildren" not in properties:
```
Comment (davesrocketshop) [inline src/Mod/Draft/draftviewproviders/view_layer.py:73]: These are the only functional changes. OverrideShapeColorChildren was renamed OverrideShapeAppearanceChildren to more accurately reflect the change in the underlying attribute.

**Review by davesrocketshop — COMMENTED** [review]
It's not clear why this reversion is necessary. Is there an issue associated with this? There may be better alternatives that allow for the continued transition from ShapeColor to ShapeAppearance

**Comment by Roy-043** [comment]
Implementing the new material handling in Draft is a delicate operation. The code changes in #13294 are either non-functional (changing formatting or changing a comment without changing the related code) or break existing functionality (changing a layer color is no longer propagated to objects grouped in that layer). There is also the conversion of old objects that needs to be addressed. Not to mention several GUI related aspects that need to be considered (for example the task panel of the [Draft_SetStyle](https://wiki.freecad.org/Draft_SetStyle) command probably has to be modified). Etc...

This PR is not about blocking a change, but about about reverting changes that currently do not make sense. I am putting this at the top of my ToDo list as I think this should be implemented in the Draft WB in the current dev cycle. But it should be done properly.

**Comment by davesrocketshop** [comment]
> Implementing the new material handling in Draft is a delicate operation. The code changes in #13294 are either non-functional (changing formatting or changing a comment without changing the related code) or break existing functionality (changing a layer color is no longer propagated to objects grouped in that layer). There is also the conversion of old objects that needs to be addressed. Not to mention several GUI related aspects that need to be considered (for example the task panel of the [Draft_SetStyle](https://wiki.freecad.org/Draft_SetStyle) command probably has to be modified). Etc...
> 
> This PR is not about blocking a change, but about about reverting changes that currently do not make sense. I am putting this at the top of my ToDo list as I think this should be implemented in the Draft WB in the current dev cycle. But it should be done properly.

Fair enough. Let me know if I can help in any way. 

FYI, there are more changes coming, including updated task panels, etc. These are not part of the Draft WB initially but may be good models for updating task panels there. I can help with that

**Comment by Roy-043** [comment]
Thanks. I going to merge this. I'll start a discussion regarding the implementation of Material in Draft.

**Comment by Roy-043** [comment]
Discussion: https://github.com/FreeCAD/FreeCAD/discussions/13657

**Pull request merged**